modules/slo-pipeline/code/main.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from slo_generator import compute
+import json
+import base64
+
+with open("exporters.json") as f:
+    exporters = json.load(f)
```

Comment:
Are there any cases where the json cannot be deserialized, and you would need to trap the resulting exception?

Reply:
Nope, it's better to let the exception show up straight in the logs. The JSON file here is configured via Terraform, so probably Terraform would catch it before deploying the Cloud Function.

Reply:
Hm, wrapping exceptions and making them more meaningful for users (eg using `SystemExit` or `sys.exit` with precise messaging) is not a bad practice.

Reply:
Just tested with a bad JSON, here is the exception:

```
>>> yaml.safe_load("{'json': 'bad_json}")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/Library/Python/2.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/Library/Python/2.7/site-packages/yaml/constructor.py", line 43, in get_single_data
    node = self.get_single_node()
  File "/Library/Python/2.7/site-packages/yaml/composer.py", line 35, in get_single_node
    if not self.check_event(StreamEndEvent):
  File "/Library/Python/2.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/Library/Python/2.7/site-packages/yaml/parser.py", line 143, in parse_implicit_document_start
    StreamEndToken):
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 247, in fetch_more_tokens
    return self.fetch_single()
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 652, in fetch_single
    self.fetch_flow_scalar(style='\'')
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 666, in fetch_flow_scalar
    self.tokens.append(self.scan_flow_scalar(style))
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 1157, in scan_flow_scalar
    chunks.extend(self.scan_flow_scalar_spaces(double, start_mark))
  File "/Library/Python/2.7/site-packages/yaml/scanner.py", line 1245, in scan_flow_scalar_spaces
    "found unexpected end of stream", self.get_mark())
yaml.scanner.ScannerError: while scanning a quoted scalar
  in "<string>", line 1, column 10:
    {'json': 'bad_json}
             ^
found unexpected end of stream
  in "<string>", line 1, column 20:
    {'json': 'bad_json}
```

It's fairly precise what the problem is, whereas if I catch it we might lose the important details (line number + position in the JSON)

Reply:
Depends, you could use the exception message as your error message, and avoid the long traceback which isn't adding a whole lot of user info, but makes the output harder to read.

I'm fine either way, so I'll resolve and either wrap or leave as is. :)